src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
     (ActorFunc)EnEncount3_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_060009A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD058.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD084.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD194.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD1EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Draw.s")
+s32 D_809AD810 = false;
+
+void EnEncount3_Init(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+
+    this->unk14A = ENCOUNT3_GET_SPAWN_INDEX(thisx);
+    this->childParams = ENCOUNT3_GET_PARAM_F80(thisx);
+    this->switchFlag = ENCOUNT3_GET_SWITCHFLAG(thisx);
+    this->unk15A = this->actor.csId;
+    this->unk16C = ENCOUNT3_GET_ROTZ(thisx) * 10.0f;
+    if (this->unk16C < 10.0f) {
+        this->unk16C = 10.0f;
+    } else if (this->unk16C > 1000.0f) {
+        this->unk16C = 1000.0f;
+    }
+    if (this->switchFlag == 0x7F) {
+        this->switchFlag = -1;
+    }
+    if (this->switchFlag >= 0 && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_Kill(&this->actor);
+    }
+    this->actor.flags |= ACTOR_FLAG_CANT_LOCK_ON;
+    this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+    func_809AD058(this);
+}
+
+void EnEncount3_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_809AD058(EnEncount3* this) {
+    this->childActorId = ACTOR_EN_JSO;
+    this->unk150 = 1;
+    this->unk15E = 0x1E;
+    this->actionFunc = func_809AD084;
+}
+
+void func_809AD084(EnEncount3* this, PlayState* play) {
+    if (this->switchFlag >= 0 && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_Kill(&this->actor);
+    } else if (!(this->unk16C < this->actor.xzDistToPlayer) && (Player_GetMask(play) == PLAYER_MASK_GARO) &&
+               !D_809AD810) {
+        if (this->unk15E > 0) {
+            this->unk15E--;
```

Comment:
`unk15E` can be called `timer` if you'd like, which it should also be in decimal i.e. `this->unk15E = 0x1E;` above can be `this->timer = 30;`